explore_simulation_approach/3_evaluate.py

```diff
@@ -0,0 +1,437 @@
+#
+# source(paste0(base_dir, '/generic_expression_patterns_modules/DE_analysis.R'))
+#
+# # File created: "<local_dir>/DE_stats/DE_stats_template_data_<project_id>_real.txt"
+# if (de_method == "deseq"){
+#     get_DE_stats_DESeq(
+#         template_DE_grouping_filename,
+#         project_id,
+#         processed_template_filename,
+#         "template",
+#         local_dir,
+#         "real"
+#     )
+# }
+# else{
+#     get_DE_stats_limma(
+#         template_DE_grouping_filename,
+#         project_id,
+#         processed_template_filename,
+#         "template",
+#         local_dir,
+#         "real"
+#     )
+# }
+
+# + magic_args="-i template_DE_grouping_filename -i project_id -i base_dir -i local_dir -i num_runs -i de_method" language="R"
+#
+# source(paste0(base_dir, '/generic_expression_patterns_modules/DE_analysis.R'))
+#
+# # Files created: "<local_dir>/DE_stats/DE_stats_simulated_data_<project_id>_<n>.txt"
+# for (i in 0:(num_runs-1)){
+#     simulated_data_filename <- paste(
+#         local_dir,
+#         "pseudo_experiment/selected_simulated_data_",
+#         project_id,
+#         "_",
+#         i,
+#         "_processed.txt",
+#         sep = ""
+#     )
+#     if (de_method == "deseq"){
+#         get_DE_stats_DESeq(
+#             template_DE_grouping_filename,
+#             project_id,
+#             simulated_data_filename,
+#             "simulated",
+#             local_dir,
+#             i
+#             )
+#     }
+#     else {
+#         get_DE_stats_limma(
+#             template_DE_grouping_filename,
+#             project_id,
+#             simulated_data_filename,
+#             "simulated",
+#             local_dir,
+#             i
+#             )
+#         }
+#     }
+
+# +
+analysis_type = "DE"
+template_DE_stats_filename = os.path.join(
+    local_dir, "DE_stats", f"DE_stats_template_data_{project_id}_real.txt"
+)
+
+# Added
+if de_method == "deseq":
+    logFC_name = "log2FoldChange"
+    pvalue_name = "padj"
+else:
+    logFC_name = "logFC"
+    pvalue_name = "adj.P.Val"
+
+template_DE_stats, simulated_DE_summary_stats = ranking.process_and_rank_genes_pathways(
+    template_DE_stats_filename,
+    local_dir,
+    num_runs,
+    project_id,
+    analysis_type,
+    col_to_rank_genes,
+    logFC_name,
+    pvalue_name,
+)
+
+# +
+# Get summary table
+summary_gene_ranks = ranking.generate_summary_table(
+    template_DE_stats_filename,
+    template_DE_stats,
+    simulated_DE_summary_stats,
+    col_to_rank_genes,
+    local_dir,
+    "gene",
+    params,
+)
+
+summary_gene_ranks.sort_values(by="Z score", ascending=False).head(10)
+# -
+
+summary_gene_ranks_sorted = summary_gene_ranks.sort_values(
+    by="Z score", ascending=False
+)
+
+# Add ranking based on Z-score
+summary_gene_ranks_sorted["rank"] = summary_gene_ranks_sorted["Z score"].rank(
+    ascending=True
+)
+
+summary_gene_ranks_sorted.head(10)
+
+# ## Traditional DE
```

Comment:
I'll add text here too. Basically we start with a template experiment that is a perturbation experiment where we declared some samples to be perturbed and others to be controls. So for the DE analysis, we are comparing those two groups of samples. In this template experiment we already know which genes are generic and which are specific, because we selected those genes when we simulated the template experiment. Now what we want to see is how highly ranked (based on logFC) the known specific genes are vs the generic genes to determine how "well' DE can distinguish between the two.